HeroVault/HeroVault/Features/User/UserFavouritesScreen/UserFavouritesScreenViewModel.swift

```diff
@@ -47,18 +47,29 @@ class UserFavouritesScreenViewModel: ViewModelDelegate {
     func reloadView() {
     }
     
-    func createImage(marvelDataIndex: Int) -> String {
-        guard var imageName = (marvelDataList?[marvelDataIndex].thumbnail) else { return "" }
-        imageName.append("/portrait_incredible.jpg")
-        return imageName.convertToHttps()
+    struct CharacterImage {
+        let imageURL: String
+        let imageName: String
     }
     
-    func createLabel(marvelDataIndex: Int) -> String {
-        guard let imageLabel = (marvelDataList?[marvelDataIndex].name) else { return "" }
-        return imageLabel
+    func marvelDataItem(marvelDataIndex: Int) -> CharacterImage? {
```

Comment:
This is a bit worrying. All of these functions deal with a marvelData Type, why do you want to specifically return a CharacterImage Type. What about a Comic Image?